### PR TITLE
Reduce the default size of max header sizes

### PR DESCRIPTION
**Pull request by maskit**

### CODE REVIEW

**Comment by maskit** [comment]
[approve ci autest]

**Comment by JosiahWI** [comment]
AuTest `proxy_serve_stale_dns_fail` failed.

**Comment by JosiahWI** [comment]
[approve ci autest]

File: doc/admin-guide/files/records.yaml.en.rst [hunk 1]
```diff
@@ -4454,12 +4454,11 @@ HTTP/2 Configuration
    Dynamic Table, however, headers still can be encoded as indexable
    representations. The upper limit is 65536.
 
-.. ts:cv:: CONFIG proxy.config.http2.max_header_list_size INT 131072
+.. ts:cv:: CONFIG proxy.config.http2.max_header_list_size INT 32768
    :reloadable:
 
    This advisory setting informs a peer of the maximum size of header list
-   that the sender is prepared to accept blocks. The default value, which is
-   the unsigned int maximum value in |TS|, implies unlimited size.
```
Comment (bneradt) [inline doc/admin-guide/files/records.yaml.en.rst:4462]: Was this default statement simply incorrect?

Comment (maskit) [reply doc/admin-guide/files/records.yaml.en.rst:4462]: Yes, the sentence was not updated on #5319.

**Comment by maskit** [comment]
There's no blockers, right?

**Review by bryancall — APPROVED** [review]

**Review by bryancall — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**